zerver/test_bugdown.py

```diff
@@ -175,6 +175,11 @@ def replaced(payload, url, phrase=''):
             converted = bugdown_convert(inline_url)
             self.assertEqual(match, converted)
 
+    def test_inline_file(self):
+        msg = 'Check out this file file:///Volumes/myserver/Users/Shared/pi.py'
+        converted = bugdown_convert(msg)
+        self.assertEqual(converted, '<p>Check out this file <a href="file:///Volumes/myserver/Users/Shared/pi.py" target="_blank" title="file:///Volumes/myserver/Users/Shared/pi.py">file:///Volumes/myserver/Users/Shared/pi.py</a></p>')
+
```

Comment:
I think what's going on is effectively a form of caching -- the bugdown library only is initialized once per realm and the code involved is initialization code.  See `make_md_engine` for details.  I think you could fix this by first creating a new empty realm with domain e.g. "file_links_test.example.com" and then using `bugdown.convert(text ,realm_domain="file_links_test.example.com")` rather than the `bugdown_convert` helper.